oak_runtime/src/node/mod.rs

```diff
@@ -131,24 +147,32 @@ impl NodeFactory<NodeConfiguration> for ServerNodeFactory {
                 Ok(Box::new(crypto::CryptoNode::new(node_name)))
             }
             Some(ConfigType::LogConfig(LogConfiguration {})) => {
-                Ok(Box::new(logger::LogNode::new(node_name)))
+                if self.permissions_configuration.allow_log_nodes {
```

Comment:
I think these checks could be moved into the `new` functions. I can think of arguments for both approaches. WDYT @tiziano88?

Reply:
Actually an alternative option would be to check the permission upfront by passing the NodeConfiguration object to a function (rather than just the specific gRPC configs). 

```rust
if !self.permission_configuration.allowed_creation(&node_configuration) {
  return Err(...);
}
match &node_configuration.config_type {
  ...
}
```